src/main/java/io/github/opencubicchunks/cubicchunks/core/asm/mixin/core/common/MixinChunk_Cubes.java

```diff
@@ -702,7 +717,7 @@ public void getEntitiesWithinAABBForEntity_CubicChunks(@Nullable Entity entityIn
     private void getPrecipitationHeight_CubicChunks_Replace(BlockPos pos, CallbackInfoReturnable<BlockPos> cbi) {
         if (isColumn) {
             // TODO: precipitationHeightMap
-            BlockPos ret = new BlockPos(pos.getX(), getHeightValue(blockToLocal(pos.getX()), blockToLocal(pos.getZ())), pos.getZ());
+            BlockPos ret = new BlockPos(pos.getX(), getHeightValue(blockToLocal(pos.getZ()), blockToLocal(pos.getZ())), pos.getZ());
```

Comment:
getZ() two times.

Reply:
That would explain some issues with rain